bitcoin/src/blockdata/script/borrowed.rs

```diff
@@ -318,6 +318,65 @@ impl Script {
         crate::Amount::from_sat(sats)
     }
 
+    /// Get the sigop count for this Script.
+    ///
+    /// The `accurate` argument decides whether to count OP_CHECKMULTISIG* as 20 sigops
+    /// or the preceeding OP_{N} N amount (the number of pubkeys in the multisig).
+    ///
+    /// Bitcoin Core uses `accurate` = false for any sig ops directly placed in the
+    /// scriptSig of any input or scriptPubkey of any output.
+    ///
+    /// Alternatively, `accurate` = true is used for redeemScripts(P2SH) and witnessScripts(P2WSH).
+    ///
+    /// (Note: taproot scripts don't count toward the sigop count of the block,
+    /// nor do they have CHECKMULTISIG operations. This function does not count OP_CHECKSIGADD,
+    /// so do not use this to try and estimate if a taproot script goes over the sigop budget.)
+    pub fn get_sigop_count(&self, accurate: bool) -> Result<usize, super::Error> {
+        const OP_INVALIDOPCODE_U8: u8 = OP_INVALIDOPCODE.to_u8();
+        const OP_PUSHNUM_1_U8: u8 = OP_PUSHNUM_1.to_u8();
+        const OP_PUSHNUM_16_U8: u8 = OP_PUSHNUM_16.to_u8();
+        // Used to get the pushed number as a u8 from PUSHNUM
+        const OP_PUSHNUM_BASE_U8: u8 = OP_PUSHNUM_1_U8 - 1;
+
+        let mut n = 0;
+        let mut last_opcode = OP_INVALIDOPCODE_U8;
+        for inst in self.instructions() {
+            match inst? {
+                Instruction::Op(opcode) => {
+                    match opcode {
+                        OP_CHECKSIG | OP_CHECKSIGVERIFY => {
+                            n += 1;
+                        }
+                        OP_CHECKMULTISIG | OP_CHECKMULTISIGVERIFY => {
+                            if accurate
+                                && (OP_PUSHNUM_1_U8..=OP_PUSHNUM_16_U8).contains(&last_opcode)
+                            {
+                                // Add the number of pubkeys in the multisig as sigop count
+                                n += (last_opcode - OP_PUSHNUM_BASE_U8) as usize;
```

Comment:
This looks wrong, if `last_opcode` is `OP_INVALIDOPCODE` this adds 175 to `n`. Shouldn't it return error?

Reply:
`(OP_PUSHNUM_1_U8..=OP_PUSHNUM_16_U8).contains(&last_opcode)` guarantees that `last_opcode` is between OP_1 and OP_16.

Reply:
:man_facepalming: that's what I get for reviewing when tired, sorry for the noise.